### PR TITLE
feat(Interactions): Added Interactions component …

### PR DESCRIPTION
**Pull request by nickmanks**
and some basic interaction children components, `Defaults` and `DragRotate`.

User can now include an Interactions component as a child of the Map context, whose children will be the individual interactions the user wishes to include:

```html
<Map>
  <View />
  <Interactions>
     <Defaults altShiftDragRotate={false} />
     <DragRotate condition={altKeyOnly} />
  </Interactions>
</Map>
```

Closes nearmap/olr#34

### CODE REVIEW

File: src/interaction/defaults.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    zoomDelta: PropTypes.number,
+    zoomDuration: PropTypes.number
+  }
+
+  constructor(props) {
+    super(props);
+
+    const {interactions} = props;
+
+    this.interaction = interactionDefaults({...props, interactions: undefined});
```
Comment (kollhof) [inline src/interaction/defaults.js]: You probably want to be explicit about the params being passed to `interactionDefaults`:
```javascript
const {
  altShiftDragRotate, constrainResolution, doubleClickZoom,
  keyboard, mouseWheelZoom, shiftDragZoom, dragPan, pinchRotate,
  zoomDelta, zoomDuration
} = props;

this.interaction = interactionDefaults({
  altShiftDragRotate, constrainResolution, doubleClickZoom,
  keyboard, mouseWheelZoom, shiftDragZoom, dragPan, pinchRotate,
  zoomDelta, zoomDuration
} );
```

File: src/interaction/defaults.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    interactions.extend([this.interaction]);
+  }
+
+  componentDidUpdate() {
+    const {props} = this;
+    this.interaction = interactionDefaults({...props, interactions: undefined});
```
Comment (kollhof) [inline src/interaction/defaults.js]: This does not appear to have any effect.
I cannot see how `this.interaction` is used anywhere.

File: src/interaction/defaults.test.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      shiftDragZoom: true,
+      dragPan: true,
+      pinchRotate: true,
+      zoomDelta: 1,
+      zoomDuration: 1
+    });
+    expect(interactions.getArray().length).toBe(1);
+  });
+
+  it('can update its props and the interaction will be updated', ()=> {
```
Comment (kollhof) [inline src/interaction/defaults.test.js]: Unless the interactions of the map are updated, I am not sure this is a valid use case.

File: src/interaction/defaults.js (hunk 2 above)
Comment (nickmanks) [reply src/interaction/defaults.js]: `interactions.extend([this.interaction])`? so a reference to the `this.interaction` will be passed, and if it changes then so will the interaction inside the interactions collection?

File: src/interactions.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import React, {createContext} from 'react';
```
Comment (kollhof) [inline src/interactions.js]: I wonder if that file should just be `src/interaction/index.js`

File: src/interaction/defaults.js (hunk 2 above)
Comment (kollhof) [reply src/interaction/defaults.js]: `this.interaction = ...` will assign a new object to `this.interaction` not mutate it.
There is no way for anything that only has a reference to the previous value of `this.interaction` to now have a reference to the new value.

**Pull request merged**